doc/releases.md

```diff
@@ -0,0 +1,120 @@
+# Releases
+
+## Table of Contents
+
+- [Release Philosophy](#release-philosophy)
+- [Release Flow](#release-flow)
+  - [Stage 0 - Automated Testing](#stage-0---automated-testing)
+  - [Stage 1 - Internal Testing](#stage-1---internal-testing)
+  - [Stage 2 - Community Dev Testing](#stage-2---community-dev-testing)
+  - [Stage 3 - Community Prod Testing](#stage-3---community-prod-testing)
+  - [Stage 4 - Release](#stage-4---release)
+- [Release Cycle](#release-cycle)
+  - [Patch Releases](#patch-releases)
+- [Performing a Release](#performing-a-release)
+- [Release Version Numbers](#release-version-numbers)
+  - [Pre-Releases and Release Candidates](#pre-releases-and-release-candidates)
+
+## Release Philosophy
+
+js-ipfs aims to have release every six weeks, two releases per quarter. During these 6 week releases, we go through 4 different stages that gives us the opportunity to test the new version against our test environments (unit, interop, integration), QA in our current production environment, IPFS apps (e.g. Desktop and WebUI) and with our community and _early testers_<sup>[1]</sup> that have IPFS running in production.
+
+We might expand the six week release schedule in case of:
+
+- No new updates to be added
+- In case of a large community event that takes the core team availability away (e.g. IPFS Conf, Dev Meetings, IPFS Camp, etc.)
+
+## Release Flow
+
+js-ipfs releases come in 5 stages designed to gradually roll out changes and reduce the impact of any regressions that may have been introduced. If we need to merge non-trivial<sup>[2]</sup> changes during the process, we start over at stage 0.
+
+![js-ipfs release flow cartoon](https://ipfs.io/ipfs/QmU5pwcGh38DqzLy3rK8GAHuWm2kK87oGqDAtqZYWhxjab)
+
+### Stage 0 - Automated Testing
+
+At this stage, we expect _all_ automated tests (unit, functional, integration, interop, testlab, performance, etc.) to pass.
+
+### Stage 1 - Internal Testing
+
+At this stage, we'll:
+
+1. Start a partial-rollout to our own infrastructure.
```

Comment:
Again, this is a noop for js-ipfs right now but with the addition of `/refs` and now GC we're one step closer to being able to be our own preload node.